### ISSUE TITLE
Boring from parent module with probe doesn't get a necessary :#= assignment

### ISSUE THREAD

**Issue by mwachs5** [chipsalliance/chisel#3557]
**Type of issue**: Bug Report

**Please provide the steps to reproduce the problem:**

See the test in #3556. Even if fixing the issue reported there, the test will not pass, because as shown in the picture below, we implement the single black arrow not the dotted two-part arrow:
<img width="596" alt="Screenshot 2023-09-26 at 4 50 21 PM" src="https://github.com/chipsalliance/chisel/assets/17858596/a9e72a01-11a4-4afa-82ff-5c1e836223ec">

**What is the current behavior?**
Note the bottom arrow in the picture: when we are goign straight from a bidirectional thing, and then turning it into an Input, we dont have any part of the code that does the stripping of direction and then using a `:#=` operator to drive the stripped-direction input to the bidirectional "real thing".

When there is a layer of hierarchy in the middle:
<img width="550" alt="Screenshot 2023-09-27 at 11 05 13 AM" src="https://github.com/chipsalliance/chisel/assets/17858596/54a44601-a13c-49f7-88d1-781c53ae3b9b">

we have to first create the probe, which handles the stripping and the define works without the `#:=`.

**What is the expected behavior?**

We need to not emit a `connect` from a bidirectional thing to a monodirectional thing.

This can be done by either emitting the individual fields (as in the `:#=` would handle) or by making a probe define and then read in the same layer of hierarchy.

The former is a little tricky to implement because we can't directly use the `:#=` operator in the boring APIs, we can only add commands to the secretCommands, and there is no direct translation from `:#=` to the commands we would want to add to secretCommands.

It might be simpler to just always make the probe, define, and read even if we are in the same layer of heirarchy.

**Please tell us about your environment:**

**Other Information**

**What is the use case for changing the behavior?**

Need to get code compiling. There is a possible workaround -- users can make the Wire(Output(...)) themself in the source of the tap.

**Comment by dtzSiFive** [comment]
> It might be simpler to just always make the probe, define, and read even if we are in the same layer of heirarchy.

This should be optimized away by CIRCT, if it's not LMK :+1: .  It does seem like it may simplify things a bit.

In a sense a `read(probe(X))` would become the implementation of a hypothetical `asPassive(X)` command, is that about right?

All good, but curious/interested: what's the reason/challenges preventing having `:#=` as a (sequence of) "secret command"?

**Comment by seldridge** [comment]
> All good, but curious/interested: what's the reason/challenges preventing having :#= as a (sequence of) "secret command"?

It's due to current implementation. This can be changed. The problem is that `:#=` eventually results in calls to `:=` [here](https://github.com/chipsalliance/chisel/blob/6dbbc73c793bdd1a357a4088abfe4704f23195a8/core/src/main/scala/chisel3/connectable/Connection.scala#L124C30-L124C30). This will push commands to the module's normal command area. Instead, we have a problem where we need to push arbitrary commands. However, the module is currently only allowing you to create a `secretConnection` through a [restricted API](https://github.com/chipsalliance/chisel/blob/6dbbc73c793bdd1a357a4088abfe4704f23195a8/core/src/main/scala/chisel3/RawModule.scala#L186) that only works with a LHS and RHS Data (which then pushes commands into the secret command array).

To be able to use `:#=` there needs to be a way to tell any connection what buffer to add commands to. If we're then starting to add arguments to `:#=` and friends related to what buffer to push to, I don't see the point in maintaining two separate buffers.

tl;dr: there are some infrastructure changes that have to happen.
